### ISSUE TITLE
Improve CONTRIBUTORS.md

### ISSUE THREAD

**Issue by brb** [weaveworks/weave#3323]
TL;DR fix CONTRIBUTORS.md to include a helpful information for external contributors on how to develop Weave Net.

1. Revisit its [current content](https://github.com/weaveworks/weave/blob/master/CONTRIBUTING.md) and remove/update obsolete parts.
2. Add requirements for dev tool chain to compile Weave Net (just Docker).
3. Add information how to test Weave Net locally (bootstrapping VMs, installing Kubernetes, , running smoke tests, etc).
4. The doc links to https://github.com/weaveworks/weave/wiki/WorkingOnWeave which has some out-of-date info (e.g. we no longer deploy docs to Wordpress). Fix it.
5. Mention #net-dev on the community Slack for discussing with other developers.
